### PR TITLE
Initial version of integration with the github api

### PR DESCRIPTION
**Pull request by TimvdLippe**
Connect the application with the GitHub api.
# How to test
1. Go to https://github.com/settings/tokens
2. `Generate new token`
3. Create the token and save the result (Important!)
4. Create the file `src/token.json`
5. Paste your token with double-quotes surrounded. If your token is asdf, `src/token.json` must contain `"asdf"`
6. Fire up your application

Projects are not fetched, but when you click a project it will fetch all open pull requests. Labels are fetched in a separate request (per pull request) because they are on a different end point :cry: 

Requests are cached using the `ETag` header. The corresponding GitHub documentation can be found at https://developer.github.com/guides/getting-started/#conditional-requests. Every response includes the `ETag` header. By setting the header `If-None-Match` to the `ETag` value, we could get a 304. This indicates no changes were made since the last request. 304's do NOT count for the rate limit!

If we receive a 304, we look up the previous result from a `_responseCache`.

### CODE REVIEW

File: src/projects/projects-page.html [hunk 1]
```diff
@@ -62,14 +67,17 @@
           </a>
         </template>
       </preview-list>
-      <project-pull-requests hidden$="[[!_shouldShowProjects(name, pr)]]" route="{{nameTail}}" tabindex="0" owner="[[owner.owner]]" name="[[name.name]]"></project-pull-requests>
+      <project-pull-requests hidden$="[[!_shouldShowProjects(projectTitle, pr)]]" route="{{nameTail}}" tabindex="0"
```
Comment (eanker) [inline src/projects/projects-page.html:70]: This method does not seem to work, or something it calls.
When _shouldShowProjects is returning true, it's still shown.
I tried debugging, but could not find the cause.

**Comment by eanker** [comment]
Due to the project not hidding properly the chunk view is very small.
In the beginning the pr's are not shown and the chunks are big, and then it flashes back soon and the pr's are shown and the chunk view is very small.

**Comment by eanker** [comment]
The new commit fixed the issue.

File: src/projects/projects-page.html (hunk 1 above)
Comment (TimvdLippe) [inline src/projects/projects-page.html:70]: Fixed :)

**Comment by txsmith** [comment]
Looks good! It seems to work. We'll need a face to face explanation on how this works though as this does not seem like a very trivial thing to achieve. I'll merge if you fix these two UI thingies:

screenshot from 2016-05-09 13 59 42

The bottom line is supposed to show the amount of reviewers instead of a user-id, and the date-line should contain something along the lines of `x days` instead of an actual date.

**Comment by txsmith** [comment]
Also, how can we run the rests you supplied in this PR?

**Comment by TimvdLippe** [comment]
The UI has been updated. You can run the tests with `polyserve` by visiting `./test/index.html` and `./test/ajax/authenticated-ajax.html`
